internal/fs/inode/file.go

```diff
@@ -212,6 +212,12 @@ func (f *FileInode) clobbered(ctx context.Context, forceFetchFromGcs bool, inclu
 		return
 	}
 
+	// For localFile, if object exists in GCS then it means file is clobbered.
+	if f.IsLocal() {
+		b = true
+		return
+	}
+
 	// We are clobbered iff the generation doesn't match our source generation.
 	oGen := Generation{o.Generation, o.MetaGeneration}
 	b = f.SourceGeneration().Compare(oGen) != 0
```

Comment:
Can you check if this is already taking care of the above condition we have added?